### DISCUSSION TITLE
how to debounce  on Controller rules ?

### DISCUSSION THREAD

**Discussion by vinodhum** [react-hook-form/react-hook-form#2995] [Q&A]
Hi,
I am trying to debounce validation onChange but rules accepts object not function
how can I delay validation using controller ?

Thank You

**Comment by bluebill1049** [comment]
`register({validate: debounce(callback)})`?

**Reply by vinodhum** [reply]
here is what I treid, imported debounce from lodash
` rules= { ...validation, validate: debounce(isUnique,1000), pattern }`
its not working

**Reply by bluebill1049** [reply]
codesandbox?
